src/daemon/rpmostreed-transaction-types.c

```diff
@@ -1389,3 +1390,169 @@ rpmostreed_transaction_new_refresh_md (GDBusMethodInvocation *invocation,
   return (RpmostreedTransaction *) self;
 
 }
+
+/* ================================KernelArg================================ */
+
+typedef struct {
+  RpmostreedTransaction parent;
+  char  *osname;
+  char **kernel_args_added;
+  char **kernel_args_deleted;
+  char **kernel_args_replaced;
+  RpmOstreeTransactionKernelArgFlags flags;
+} KernelArgTransaction;
+
+typedef RpmostreedTransactionClass KernelArgTransactionClass;
+
+GType kernel_arg_transaction_get_type (void);
+
+G_DEFINE_TYPE (KernelArgTransaction,
+               kernel_arg_transaction,
+               RPMOSTREED_TYPE_TRANSACTION)
+
+static void
+kernel_arg_transaction_finalize (GObject *object)
+{
+  KernelArgTransaction *self;
+
+  self = (KernelArgTransaction *) object;
+  g_free (self->osname);
+  g_strfreev (self->kernel_args_added);
+  g_strfreev (self->kernel_args_deleted);
+  g_strfreev (self->kernel_args_replaced);
+
+  G_OBJECT_CLASS (kernel_arg_transaction_parent_class)->finalize (object);
+}
+
+static gboolean
+kernel_arg_transaction_execute (RpmostreedTransaction *transaction,
+                                GCancellable *cancellable,
+                                GError **error)
+{
+  KernelArgTransaction *self = (KernelArgTransaction *) transaction;
+  OstreeSysroot *sysroot = rpmostreed_transaction_get_sysroot (transaction);
+
+  /* Set this boolean to have a default value */
+  gboolean import_from_cmd = FALSE;
+
+  if (self->flags & RPMOSTREE_TRANSACTION_KERNEL_ARG_FLAG_IMPORT_CMD)
```

Comment:
I read "import cmd" as an abbreviation for "import command" - let's say "_IMPORT_CMDLINE" to be clearer that it's the "command line"?  Or maybe to be even more clear, say "FLAG_IMPORT_PROC_CMDLINE" ?  And name the boolean `import_from_proc_cmdline`.